app/forms/candidate_interface/account_recovery_form.rb

```diff
@@ -0,0 +1,44 @@
+module CandidateInterface
+  class AccountRecoveryForm
+    include ActiveModel::Model
+
+    attr_accessor :code
+    attr_reader :valid_account_recovery_request, :current_candidate, :old_candidate
+
+    validates :code, presence: true
+    validate :account_recovery, unless: -> { valid_account_recovery_request && old_candidate }
+    validate :previous_account_has_no_one_login, if: -> { valid_account_recovery_request && old_candidate }
+
+    def initialize(current_candidate:, code: nil)
+      self.code = code
+      @current_candidate = current_candidate
+    end
+
+    def call
+      @valid_account_recovery_request = AccountRecoveryRequest.where(code:)
+        .where('created_at >= ?', 1.hour.ago).first
+      @old_candidate = Candidate.find_by(email_address: valid_account_recovery_request&.previous_account_email)
```

Comment:
Yeah that's a good point, I was hoping that this would always generate a unique code 
https://github.com/DFE-Digital/apply-for-teacher-training/blob/cbdf7039273d3b43e96f2f65cc9782e572ab6d0a/app/models/account_recovery_request.rb#L9-L13

Reply:
Yes! Thanks for pointing that out. I should read the whole thing before I comment!

That said, I'm not sure this will work, or work efficiently, if we are encrypting the codes. It's something to keep in mind.

Reply:
Yeah, good point. I haven't thought about encrypting the code, I'll do some testing when I'll create the production PR 👍🏻